### PR TITLE
Innermost loop check

### PR DESCRIPTION
**Pull request by definelicht**
Make the check whether a loop is innermost more sophisticated by taking nested SDFGs and unrolled maps into account.

@manuelburger

### CODE REVIEW

**Comment by definelicht** [comment]
@manuelburger This fixes the extra pipelines in the SDFG you sent

File: dace/codegen/targets/fpga.py [hunk 1]
```diff
@@ -880,6 +880,22 @@ def copy_memory(self, sdfg, dfg, state_id, src_node, dst_node, edge,
     def _generate_PipelineEntry(self, *args, **kwargs):
         self._generate_MapEntry(*args, **kwargs)
 
+    def _is_innermost(self, scope, scope_dict):
+        to_search = list(scope)
+        while len(to_search) > 0:
+            x = to_search.pop()
+            if (isinstance(x, dace.graph.nodes.MapEntry)
+                    or isinstance(x, PipelineEntry)):
```
Comment (tbennun) [inline dace/codegen/targets/fpga.py:888]: Check for `EntryNode` or at least isinstance with a tuple as a second argument.

Comment (definelicht) [reply dace/codegen/targets/fpga.py:888]: I want to be explicit here, it should only be an entry node that I know for sure that we would pipeline. But I'll add the tuple

Comment (definelicht) [reply dace/codegen/targets/fpga.py]: Fixed

**Pull request merged**